src/Adapter/MSTestAdapter.PlatformServices/Services/ThreadOperations.cs

```diff
@@ -31,16 +66,13 @@ public bool Execute(Action action, int timeout, CancellationToken cancelToken)
             Name = "MSTest Test Execution Thread",
         };
 
-        // This is not ideal and should be replaced with a mechanism that use STA thread only for tests that need it
-        // (based on user configuration). We are currently relying on testhost to be STA or MTA and will run ALL tests
-        // in this mode which is not ideal for different hosting mechanism not for performance.
         executionThread.SetApartmentState(Thread.CurrentThread.GetApartmentState());
         executionThread.Start();
 
         try
         {
-            var executionTask = Task.Run(() => executionThread.Join(timeout), cancelToken);
-            if (executionTask.Wait(timeout, cancelToken))
+            var executionTask = Task.Run(() => executionThread.Join(timeout), cancellationToken);
+            if (executionTask.Wait(timeout, cancellationToken))
```

Comment:
I think we can use only the `cancellationToken` here and remove the timeout, the wait will:
* throw for exception
* return true and the result can be true/false(expired)
* throw cancelled